iceoryx_utils/include/iceoryx_utils/cxx/helplets.hpp

```diff
@@ -268,6 +268,15 @@ constexpr bool isCompiledOn32BitSystem()
     return INTPTR_MAX == INT32_MAX;
 }
 
+/// @brief Checks if an unsigned integer is a power of two
+/// @return true if power of two, otherwise false
+template <typename T>
+constexpr bool isPowerOfTwo(const T n)
+{
+    static_assert(std::is_unsigned<T>::value && !std::is_same<T, bool>::value, "Only unsigned integer are allowed!");
+    return n && ((n & (n - 1U)) == 0U);
```

Comment:
for the mathematicians to prove ;)

Reply:
1. n != 0 so the first part is true. 
   * if n is a power of two only one 1 (at position k) can be found in the bit representation, everything else is 0
   * in n - 1 there are all bits set to 1 which are at a position less then k.
   * example: `1 0 0 0 & 0 1 1 1 = 0 0 0 0 == 0U`
2. n == 0 first part is false ... all fine

Reply:
We already had this @elBoberido and it works.
The above is no full proof, but here you go.
Assume n != 0, it is handled in the first operand of &&.
If is a power of two it is trivial since exactly one bit is 1 and after subtraction it is 0 and the lower bits are 1 and so the result of & is 0.
Otherwise at least two bits are 1 and subtraction of 1 will leave the highest bit unchanged, causing the result of & to be !=0. (subtraction will change all the bits lower than the lowest bit set to 1 and the lowest set bit to 0 but no more)

A related task (more general):
Now count the number of set bits k in a n bit number but the algorithm only is allowed to take time O(k), i.e. O(n) is not sufficient (and does not need much insight, hence is boring).